agent/connect/ca/provider_vault.go

```diff
@@ -864,6 +866,36 @@ func (v *VaultProvider) setNamespace(namespace string) func() {
 	}
 }
 
+// autotidyIssuers sets Vault's auto-tidy to remove expired issuers
+// Returns a boolean on success for testing (as there is no post-facto way of
+// checking if it is set). Logs at info level on failure to set and why, returning the
+// log message for test purposes as well.
+func (v *VaultProvider) autotidyIssuers(path string) (bool, string) {
+	s, err := v.client.Logical().Write(path+"/config/auto-tidy",
+		map[string]interface{}{
+			"enabled":              true,
+			"tidy_expired_issuers": true,
+		})
+	if err != nil {
+		errStr := err.Error()
+		switch {
+		case strings.Contains(errStr, "404"):
+			errStr = "vault versions < 1.12 don't support auto-tidy"
+		case strings.Contains(errStr, "400"):
+			errStr = "vault versions < 1.13 don't support the tidy_expired_issuers field"
+		case strings.Contains(errStr, "403"):
+			errStr = "permission denied on auto-tidy path in vault"
+		}
+		v.logger.Info("Unable to enable Vault's auto-tidy feature for expired issuers", "reason", errStr, "path", path)
+		return false, errStr
```

Comment:
Do we need these return values any more?

Reply:
Hmm... the code below duplicates that output, so those aren't required. I thought it made it more explicit but looking at it again I think it might actually be clearer to remove the early return and add a comment to block off the bottom bit as testing related.

Reply:
It wasn't quite as clean as I'd hoped but I went ahead. I'll wait on merging for a bit in case you hate the revision.